docs: Clarifying that Bicep samples replace the usual .json parameter file (#290)
Azure/bicep-registry-modules: modules/lz/sub-vending/README.md

> See the different types of landing zones in the Azure Landing Zones documentation here: [What is an Azure landing zone? - Platform vs. application landing zones](https://learn.microsoft.com/azure/cloud-adoption-framework/ready/landing-zone/#platform-vs-application-landing-zones)

> The modules are designed to be instantiated many times, once for each desired landing zone. Since the entire logic is captured in the registry's module, you may use the below proposed Bicep code samples to be the source of parameters instead of the usual parameters.json file, and have in this case one Bicep file per landing zone. If you decide instead to use a parameters.json (as the one proposed below), you will need to declare parameters in the Bicep file so it can fetch them from the JSON file.

Reviewer: ```suggestion
The modules are designed to be instantiated many times, once for each desired landing zone. Since the logic is captured in the registry's module, you may use the below proposed Bicep code samples to be the source of parameters instead of the usual `parameters.json` file, and have in this case one Bicep file per landing zone. If you decide instead to use a `parameters.json` (as the one proposed below), you will need to declare parameters in the Bicep file so it can accept them as parameter inputs from the parameters JSON file.
```

### Example JSON Parameter File

> These parameters would need to be declared in the Bicep file using them (currently not present in the above provided Bicep samples). Note that it is not necessary to use JSON files here since in the proposed design pattern, the Bicep file would be the source of parameters.

Reviewer: ```suggestion
These parameters would need to be declared in the Bicep file using them (currently not present in the above provided Bicep samples). Note that it is not necessary to use JSON parameter files here since in the proposed design pattern, the Bicep file would be the source of parameters.
```

> These parameters would need to be declared in the Bicep file using them (currently not present in the above provided Bicep samples). Note that it is not necessary to use JSON parameter files here since in the proposed design pattern, the Bicep file would be the source of parameters.

Reviewer 2: I understand the other examples show how to hardcode the values directly into the module, which avoids the new for a parameters file. But I still find myself using a parameters.json pretty frequently. Very particularly when deploying a bicep file from VS Code.

I think it's great you have included the parameters.json example, but I think it would be best to just also include here an example of the .bicep file that uses it.

In the example, the parameters can be greatly reduced (compared to main.bicep) and be something like this (replacing '...' with the rest of the params, which FYI GitHub Copilot does a great job helping to auto-complete)

```bicep
param subscriptionDisplayName string
param subscriptionAliasName string
param subscriptionBillingScope string
@allowed([ 'DevTest', 'Production' ])
param subscriptionWorkload string = 'Production'
param existingSubscriptionId string = ''
param subscriptionManagementGroupAssociationEnabled bool = true
...

module sub005 'br/public:lz/sub-vending:1.2.2' = {
  name: 'sub005'
  params: {
    subscriptionDisplayName: subscriptionDisplayName
    subscriptionAliasName: subscriptionAliasName
    subscriptionBillingScope: subscriptionBillingScope
    subscriptionWorkload: subscriptionWorkload 
    existingSubscriptionId: existingSubscriptionId 
    subscriptionManagementGroupAssociationEnabled: subscriptionManagementGroupAssociationEnabled
    ...
  }
}

output subscriptionId string = sub005.outputs.subscriptionId
output subscriptionResourceId string = sub005.outputs.subscriptionResourceId
```

Reviewer 2: ```suggestion
These parameters need to be declared in the Bicep file using them as shown in this example. Note that it is not necessary to use JSON parameter files here since in the proposed design pattern, the Bicep file would be the source of parameters.

```bicep
param subscriptionDisplayName string
param subscriptionAliasName string
param subscriptionBillingScope string
@allowed([ 'DevTest', 'Production' ])
param subscriptionWorkload string = 'Production'
param existingSubscriptionId string = ''
param subscriptionManagementGroupAssociationEnabled bool = true
...

module sub005 'br/public:lz/sub-vending:1.2.2' = {
  name: 'sub005'
  params: {
    subscriptionDisplayName: subscriptionDisplayName
    subscriptionAliasName: subscriptionAliasName
    subscriptionBillingScope: subscriptionBillingScope
    subscriptionWorkload: subscriptionWorkload 
    existingSubscriptionId: existingSubscriptionId 
    subscriptionManagementGroupAssociationEnabled: subscriptionManagementGroupAssociationEnabled
    ...
  }
}

output subscriptionId string = sub005.outputs.subscriptionId
output subscriptionResourceId string = sub005.outputs.subscriptionResourceId
```
```
